src/ORM/Association/HasMany.php

```diff
@@ -117,6 +165,10 @@ public function saveAssociated(EntityInterface $entity, array $options = [])
                 $targetEntity->set($properties, ['guard' => false]);
             }
 
+            if ($this->_saveStrategy === self::SAVE_REPLACE) {
+                $targetEntity->isNew(true);
```

Comment:
They should do, because the primary key attribute is has to be present in the attribute list.